# Profitable App Profiles for the App Store and Google Play Markets

Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets. We're working as data analysts for a company that builds Android and iOS mobile apps, and our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

# Opening data

In [2]:
open_app = open('AppleStore.csv')
from csv import reader
read_app = reader(open_app)
apps_data = list(read_app)

In [3]:
open_app = open('googleplaystore.csv')
from csv import reader
read_app = reader(open_app)
google_data = list(read_app)

# Exploring data

In [4]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [5]:
explore_data(google_data, 0, 2, rows_and_columns = True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Number of rows: 10842
Number of columns: 13


In [6]:
explore_data(apps_data, 0,2, rows_and_columns = True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 7198
Number of columns: 16


# Deleting wrong data

In [7]:
print(google_data[10473])
print(len(google_data))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10842


In [8]:
del google_data[10473]

In [9]:
print(len(google_data))

10841


# Removing duplicated entries

After reviewing data, its apparent that there are duplicated apps in android under different genre.

## Part one

In [10]:
duplicated_gapps = []
unique_gapps = []

for row in google_data[1:]:
    name = row[0]
    if name in unique_gapps:
        duplicated_gapps.append(name)
    else:
        unique_gapps.append(name)
        
print('Number of duplicate apps: ', len(duplicated_gapps))
print('\n')
print('Number of unique apps: ', len(unique_gapps))

Number of duplicate apps:  1181


Number of unique apps:  9659


In [11]:
duplicated_aapps = []
unique_aapps = []

for row in apps_data[1:]:
    name = row[0]
    if name in unique_aapps:
        duplicated_gapps.append(name)
    else:
        unique_aapps.append(name)
        
print('Number of duplicate apps: ', len(duplicated_aapps))
print('\n')
print('Number of unique apps: ', len(unique_aapps))

Number of duplicate apps:  0


Number of unique apps:  7197


In [12]:
print(duplicated_gapps[:5])

['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']


In [13]:
for row in google_data:
    name = row[0]
    if name == 'Instagram':
        print(row)
        print('\n')
    

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']




In [14]:
for row in google_data:
    name = row[0]
    if name == 'Box':
        print(row)
        print('\n')

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']




The total 1181 duplicated apps' information were some exactly same but some have different review numbers which is probably due to the fact that data collection points within different time frame. Hence when removing those duplicated apps, we will not remove randomly but criterion of the lowest rating numbers.

## Part two

In [15]:
reviews_max = {}
for row in google_data[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))        

9659


In [16]:
android_clean = []
already_added = []

for row in google_data[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(row)
        already_added.append(name)
print("Total rows of android data: ", len(android_clean))

Total rows of android data:  9659


So, after cleaning the android data, total rows is matching unique_data i captured previously.

# Removing Non-English apps

## Part one

There are some apps were identified as non-english and they are needs to be filtered out from the dataset.

In [17]:
print(apps_data[813][1])
print(apps_data[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

BATTLE BEARS -1
Beast Poker
中国語 AQリスニング
لعبة تقدر تربح DZ


In [18]:
def english_check(string):
    for letters in string:
        if ord(letters) > 127:
            return False
        else:
            return True
        
string = 'Instagram'
print(english_check(string))
string = '爱奇艺PPS -《欢乐颂2》电视剧热播'
print(english_check(string))
string = 'Docs To Go™ Free Office Suite'
print(english_check(string))
string = 'Instachat 😜'
print(english_check(string))
string = 'لعبة تقدر تربح DZ'
print(english_check(string))
string = 'Сайн уу'
print(english_check(string))

True
False
True
True
False
False


## Part two

Assuming some names could have certain characters that may fall out of 127 and causing unnecessary filtering out, new modification will be added to the function checking if name is english or not.

In [19]:
def english_check(string):
    failed = 0
    for letters in string:
        if ord(letters) > 127:
            failed += 1
            
    if failed > 3: 
        return False
    else:
        return True

string = 'Docs To Go™ Free Office Suite'
print(english_check(string))
string = 'Instachat 😜'
print(english_check(string))
string = '爱奇艺PPS -《欢乐颂2》电视剧热播'
print(english_check(string))

True
True
False


In [20]:
android_eng = []
ios_eng = []

for row in android_clean:
    name = row[0]
    if english_check(name):
        android_eng.append(row)
        
for row in apps_data[1:]:
    name = row[1]
    if english_check(name):
        ios_eng.append(row)

In [21]:
explore_data(android_eng, 0, 2, rows_and_columns = True)
print('\n')
explore_data(ios_eng, 0, 2, rows_and_columns = True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 6183
Number of columns: 16


So, from android apps 45 and from ios total of 1014 apps were removed from the cleaned data set.

In [22]:
print("Numbers of non-english android apps: ", len(android_clean) - len(android_eng))
print('\n')
print("Numbers of non-english ios apps: ", len(apps_data) - len(ios_eng))

Numbers of non-english android apps:  45


Numbers of non-english ios apps:  1015


# Isolating free apps

In [23]:
free_android = []

for row in android_eng:
    price = row[6]
    if price == 'Free' or price == 0.0:
        free_android.append(row)
        
explore_data(free_android, 0, 2, rows_and_columns = True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 8863
Number of columns: 13


In [24]:
free_ios = []

for row in ios_eng:
    price = row[4]
    if price == '0.0':
        free_ios.append(row)
        
explore_data(free_ios, 0, 2, rows_and_columns = True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 3222
Number of columns: 16


Finally, we left with total of 3222 apps from ios and 8863 apps from android. 

# Most Common Apps by Genre

## Part one

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we then develop it further.
3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the 'prime_genre' column of the App Store data set, and the 'Genres' and 'Category' columns of the Google Play data set.

## Part two

We'll build two functions we can use to analyze the frequency tables:

-  One function to generate frequency tables that show percentages
-  Another function that we can use to display the percentages in a descending order

In [25]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for row in dataset:
        category = row[index]
        total += 1
        if category in table:
            table[category] += 1
        else:
            table[category] = 1
    
    freq_table = {}
    for key in table:
        percentage = (table[key] / total) * 100
        freq_table[key] = percentage
    return freq_table
    
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

## Part three

We start by examining the frequency table for the 'genre' column of the Android Store data set.

In [26]:
display_table(free_android, -4)

Tools : 8.450863138892023
Entertainment : 6.070179397495204
Education : 5.348076272142616
Business : 4.592124562789123
Productivity : 3.8925871601038025
Lifestyle : 3.8925871601038025
Finance : 3.7007785174320205
Medical : 3.5315355974275078
Sports : 3.463838429425702
Personalization : 3.317161232088458
Communication : 3.2381812027530184
Action : 3.102786866749408
Health & Fitness : 3.0802211440821394
Photography : 2.944826808078529
News & Magazines : 2.798149610741284
Social : 2.6627552747376737
Travel & Local : 2.324269434728647
Shopping : 2.245289405393208
Books & Reference : 2.1437436533904997
Simulation : 2.042197901387792
Dating : 1.8616721200496444
Arcade : 1.8503892587160102
Video Players & Editors : 1.771409229380571
Casual : 1.7601263680469368
Maps & Navigation : 1.399074805370642
Food & Drink : 1.241114746699763
Puzzle : 1.128286133363421
Racing : 0.9928917973598104
Role Playing : 0.9364774906916393
Libraries & Demo : 0.9364774906916393
Auto & Vehicles : 0.9251946293580051
S

There are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

Ios app by 'prime_genre':

In [27]:
display_table(free_ios, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Android app by 'categary':

In [28]:
display_table(free_android, 1)

FAMILY : 18.898792733837304
GAME : 9.725826469592688
TOOLS : 8.462146000225657
BUSINESS : 4.592124562789123
LIFESTYLE : 3.9038700214374367
PRODUCTIVITY : 3.8925871601038025
FINANCE : 3.7007785174320205
MEDICAL : 3.5315355974275078
SPORTS : 3.396141261423897
PERSONALIZATION : 3.317161232088458
COMMUNICATION : 3.2381812027530184
HEALTH_AND_FITNESS : 3.0802211440821394
PHOTOGRAPHY : 2.944826808078529
NEWS_AND_MAGAZINES : 2.798149610741284
SOCIAL : 2.6627552747376737
TRAVEL_AND_LOCAL : 2.335552296062281
SHOPPING : 2.245289405393208
BOOKS_AND_REFERENCE : 2.1437436533904997
DATING : 1.8616721200496444
VIDEO_PLAYERS : 1.7939749520478394
MAPS_AND_NAVIGATION : 1.399074805370642
FOOD_AND_DRINK : 1.241114746699763
EDUCATION : 1.1621347173643235
ENTERTAINMENT : 0.9590432133589079
LIBRARIES_AND_DEMO : 0.9364774906916393
AUTO_AND_VEHICLES : 0.9251946293580051
HOUSE_AND_HOME : 0.8236488773552973
WEATHER : 0.8010831546880289
EVENTS : 0.7108202640189552
PARENTING : 0.6544059573507841
ART_AND_DESIGN : 0

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

# Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the 'Installs' column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the 'rating_count_tot' app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [29]:
genre_ios = freq_table(free_ios, -5)

for genre in genre_ios:
    total = 0
    len_genre = 0
    for row in free_ios:
        genre_app = row[-5] 
        if genre_app == genre:
            ratings = float(row[5])
            total += ratings
            len_genre += 1
            
    avg_rating = total / len_genre
    print(genre, ": ", avg_rating)

Social Networking :  71548.34905660378
Photo & Video :  28441.54375
Games :  22788.6696905016
Music :  57326.530303030304
Reference :  74942.11111111111
Health & Fitness :  23298.015384615384
Weather :  52279.892857142855
Utilities :  18684.456790123455
Travel :  28243.8
Shopping :  26919.690476190477
News :  21248.023255813954
Navigation :  86090.33333333333
Lifestyle :  16485.764705882353
Entertainment :  14029.830708661417
Food & Drink :  33333.92307692308
Sports :  23008.898550724636
Book :  39758.5
Finance :  31467.944444444445
Education :  7003.983050847458
Productivity :  21028.410714285714
Business :  7491.117647058823
Catalogs :  4004.0
Medical :  612.0


After looking through the top three genres within free ios applications, the most suitable genre would be 'Music'. Because 'Navigation's half million ratings only coming from GPS and Google maps. And this field is intensely concetrantred by few competitors and would require significant amount of data collection, technical proficiency needed. For 'Reference', its mainly driven by Bible. 

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

In [30]:
for row in free_ios:
    if row[-5] == "Navigation":
        print(row[1], "; ", row[5])
        
print('\n')

for row in free_ios:
    if row[-5] == "Reference":
        print(row[1], "; ", row[5])
        
print('\n')

for row in free_ios:
    if row[-5] == "Music":
        print(row[1], "; ", row[5])

Waze - GPS Navigation, Maps & Real-time Traffic ;  345046
Google Maps - Navigation & Transit ;  154911
Geocaching® ;  12811
CoPilot GPS – Car Navigation & Offline Maps ;  3582
ImmobilienScout24: Real Estate Search in Germany ;  187
Railway Route Search ;  5


Bible ;  985920
Dictionary.com Dictionary & Thesaurus ;  200047
Dictionary.com Dictionary & Thesaurus for iPad ;  54175
Google Translate ;  26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran ;  18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition ;  17588
Merriam-Webster Dictionary ;  16849
Night Sky ;  12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) ;  8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools ;  4693
GUNS MODS for Minecraft PC Edition - Mods Tools ;  1497
Guides for Pokémon GO - Pokemon GO News and Cheats ;  826
WWDC ;  762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (M

# Most Popular Apps by Genre on the App Store

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [31]:
display_table(free_android, 5)

1,000,000+ : 15.728308699086089
100,000+ : 11.55365000564143
10,000,000+ : 10.549475346947986
10,000+ : 10.199706645605326
1,000+ : 8.394448832223853
100+ : 6.916393997517771
5,000,000+ : 6.826131106848697
500,000+ : 5.562450637481666
50,000+ : 4.772650344127271
5,000+ : 4.513144533453684
10+ : 3.542818458761142
500+ : 3.2494640640866526
50,000,000+ : 2.3017037120613786
100,000,000+ : 2.1324607920568655
50+ : 1.9180864267178157
5+ : 0.7898002933543946
1+ : 0.5077287600135394
500,000,000+ : 0.270788672007221
1,000,000,000+ : 0.2256572266726842
0+ : 0.045131445334536835


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [32]:
category_google = freq_table(free_android,1)

for category in category_google:
    total = 0
    len_category = 0
    for app in free_android:
        app_genre = app[1]
        if app_genre == category:
            n_installs = app[5]
            n_installs = n_installs.replace('+', '')
            n_installs = n_installs.replace(',', '')
            total += float(n_installs)
            len_category += 1
            
    avg_installs = total / len_category
    print(category, ": ", avg_installs)
    

ART_AND_DESIGN :  1986335.0877192982
AUTO_AND_VEHICLES :  647317.8170731707
BEAUTY :  513151.88679245283
BOOKS_AND_REFERENCE :  8767811.894736841
BUSINESS :  1712290.1474201474
COMICS :  817657.2727272727
COMMUNICATION :  38456119.167247385
DATING :  854028.8303030303
EDUCATION :  1833495.145631068
ENTERTAINMENT :  11640705.88235294
EVENTS :  253542.22222222222
FINANCE :  1387692.475609756
FOOD_AND_DRINK :  1924897.7363636363
HEALTH_AND_FITNESS :  4188821.9853479853
HOUSE_AND_HOME :  1331540.5616438356
LIBRARIES_AND_DEMO :  638503.734939759
LIFESTYLE :  1437816.2687861272
GAME :  15588015.603248259
FAMILY :  3697848.1731343283
MEDICAL :  120550.61980830671
SOCIAL :  23253652.127118643
SHOPPING :  7036877.311557789
PHOTOGRAPHY :  17840110.40229885
SPORTS :  3638640.1428571427
TRAVEL_AND_LOCAL :  13984077.710144928
TOOLS :  10801391.298666667
PERSONALIZATION :  5201482.6122448975
PRODUCTIVITY :  16787331.344927534
PARENTING :  542603.6206896552
WEATHER :  5074486.197183099
VIDEO_PLAYERS 

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [33]:
for row in free_android:
    if row[1] == "COMMUNICATION" and (row[5] == '1,000,000,000+'
                                      or row[5] == '500,000,000+'):
        print(row[0], ": ", row[5])
        
print('\n')

for row in free_android:
    if row[1] == "GAME" and (row[5] == '1,000,000,000+'
                            or row[5] == '500,000,000+'):
        print(row[0], ": ", row[5])
        
print('\n')

for row in free_android:
    if row[1] == "BOOKS_AND_REFERENCE" and (row[5] == '1,000,000,000+'
                                       or row[5] == '500,000,000+'):
        print(row[0], ": ", row[5])

WhatsApp Messenger :  1,000,000,000+
Google Duo - High Quality Video Calls :  500,000,000+
Messenger – Text and Video Chat for Free :  1,000,000,000+
imo free video calls and chat :  500,000,000+
Skype - free IM & video calls :  1,000,000,000+
LINE: Free Calls & Messages :  500,000,000+
Google Chrome: Fast & Secure :  1,000,000,000+
UC Browser - Fast Download Private & Secure :  500,000,000+
Gmail :  1,000,000,000+
Hangouts :  1,000,000,000+
Viber Messenger :  500,000,000+


Candy Crush Saga :  500,000,000+
Subway Surfers :  1,000,000,000+
Pou :  500,000,000+
My Talking Tom :  500,000,000+
Temple Run 2 :  500,000,000+


Google Play Books :  1,000,000,000+


# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.